src/common/util/queueImmut.ml

```diff
@@ -0,0 +1,126 @@
+(* A Queue (FIFO, imutable) implementation, using two Lists for efficient push and pop handling.
```

Comment:
Is this your own implementation or is it borrowed somewhere? If it is, we should credit the place (and verify licenses are compatible).

Reply:
this is my own implementation

Reply:
Ok, nice! Did you compare performance with just using the normale Queue from OCaml? Was a difference notable?

Reply:
I didn't test the performance (the "efficienct" was related to the fact that two lists are used, which should be more efficient than using one list).
The reason for the new Queue is that the Ocaml Queue is mutable. When I edit the Queue in the `enter `transfer function, I need the old queue for the caller and the new queue for the callee. With the Ocaml Queue, I would be forced to copy the queue, and then edit the copied queue. Therefore I implemented an immutable Queue (I just recognized the typo in the comment :))), with which I don't need to copy the whole queue before editing.

Reply:
Could just use `BatDeque`, which is the same thing I think.